From: Byron Bradley <byron.bbradley@gmail.com>
Subject: [PATCH] Orion: Support the Buffalo Linkstation Pro/Live Platform

The Buffalo Linkstation Pro/Live is the same hardware as the
Kurobox Pro but without the NAND flash. This patch adds a
second MACHINE_START macro to the Kurobox setup file to minimise
code duplication.

Signed-off-by: Byron Bradley <byron.bbradley@gmail.com>
Acked-by: Nicolas Pitre <nico@marvell.com>

---
 arch/arm/mach-orion/Kconfig             |    8 ++++++++
 arch/arm/mach-orion/Makefile            |    1 +
 arch/arm/mach-orion/kurobox_pro-setup.c |   25 +++++++++++++++++++------
 3 files changed, 28 insertions(+), 6 deletions(-)

--- a/arch/arm/mach-orion/Kconfig~	2008-04-20 12:55:33.000000000 +0000
+++ b/arch/arm/mach-orion/Kconfig	2008-04-20 12:56:39.000000000 +0000
@@ -23,6 +23,14 @@
 	  Say 'Y' here if you want your kernel to support the
 	  KuroBox Pro platform.
 
+config MACH_LINKSTATION_PRO
+	bool "Buffalo Linkstation Pro/Live"
+	select I2C_BOARDINFO
+	help
+	  Say 'Y' here if you want your kernel to support the
+	  Buffalo Linkstation Pro/Live platform. Both v1 and
+	  v2 devices are supported.
+
 config MACH_DNS323
 	bool "D-Link DNS-323"
 	select I2C_BOARDINFO
diff --git a/arch/arm/mach-orion/Makefile b/arch/arm/mach-orion/Makefile
index f91d937..98378dc 100644
--- a/arch/arm/mach-orion/Makefile
+++ b/arch/arm/mach-orion/Makefile
@@ -2,5 +2,6 @@ obj-y				+= common.o addr-map.o pci.o gpio.o irq.o time.o
 obj-$(CONFIG_MACH_DB88F5281)	+= db88f5281-setup.o
 obj-$(CONFIG_MACH_RD88F5182)	+= rd88f5182-setup.o
 obj-$(CONFIG_MACH_KUROBOX_PRO)	+= kurobox_pro-setup.o
+obj-$(CONFIG_MACH_LINKSTATION_PRO) += kurobox_pro-setup.o
 obj-$(CONFIG_MACH_DNS323)	+= dns323-setup.o
 obj-$(CONFIG_MACH_TS209)	+= ts209-setup.o
diff --git a/arch/arm/mach-orion/kurobox_pro-setup.c b/arch/arm/mach-orion/kurobox_pro-setup.c
index 785a07b..aece0a9 100644
--- a/arch/arm/mach-orion/kurobox_pro-setup.c
+++ b/arch/arm/mach-orion/kurobox_pro-setup.c
@@ -178,11 +178,6 @@ static struct mv_sata_platform_data kurobox_pro_sata_data = {
  * General Setup
  ****************************************************************************/
 
-static struct platform_device *kurobox_pro_devices[] __initdata = {
-	&kurobox_pro_nor_flash,
-	&kurobox_pro_nand_flash,
-};
-
 static void __init kurobox_pro_init(void)
 {
 	/*
@@ -225,12 +220,15 @@ static void __init kurobox_pro_init(void)
 
 	orion_gpio_set_valid_pins(0x0000000c);
 
-	platform_add_devices(kurobox_pro_devices, ARRAY_SIZE(kurobox_pro_devices));
+	platform_device_register(&kurobox_pro_nor_flash);
+	if (machine_is_kurobox_pro())
+		platform_device_register(&kurobox_pro_nand_flash);
 	i2c_register_board_info(0, &kurobox_pro_i2c_rtc, 1);
 	orion_eth_init(&kurobox_pro_eth_data);
 	orion_sata_init(&kurobox_pro_sata_data);
 }
 
+#ifdef CONFIG_MACH_KUROBOX_PRO
 MACHINE_START(KUROBOX_PRO, "Buffalo/Revogear Kurobox Pro")
 	/* Maintainer: Ronen Shitrit <rshitrit@marvell.com> */
 	.phys_io	= ORION_REGS_PHYS_BASE,
@@ -242,3 +240,18 @@ MACHINE_START(KUROBOX_PRO, "Buffalo/Revogear Kurobox Pro")
 	.timer		= &orion_timer,
 	.fixup		= tag_fixup_mem32,
 MACHINE_END
+#endif
+
+#ifdef CONFIG_MACH_LINKSTATION_PRO
+MACHINE_START(LINKSTATION_PRO, "Buffalo Linkstation Pro/Live")
+	/* Maintainer: Byron Bradley <byron.bbradley@gmail.com>> */
+	.phys_io	= ORION_REGS_PHYS_BASE,
+	.io_pg_offst	= ((ORION_REGS_VIRT_BASE) >> 18) & 0xFFFC,
+	.boot_params	= 0x00000100,
+	.init_machine	= kurobox_pro_init,
+	.map_io		= orion_map_io,
+	.init_irq	= orion_init_irq,
+	.timer		= &orion_timer,
+	.fixup		= tag_fixup_mem32,
+MACHINE_END
+#endif
-- 
1.5.4.GIT

